### ISSUE TITLE
Transforming datasets before or after Split?

### ISSUE THREAD

**Issue by VIGS25** [deepchem/deepchem#1556]
I was writing the load function for thermodynamic solubility datasets and noticed many of the functions in `dc.molnet` have the transformation before the dataset is split. 

In a normal workflow, the dataset is split into train-valid-test and the feature transformation, for e.g. normalization is carried out after that, to prevent any dependencies between the train-test-valid datasets. 

Should the same idea apply here as well?

**Comment by VIGS25** [comment]
Example of `load_delaney` (https://github.com/deepchem/deepchem/blob/master/deepchem/molnet/load_function/delaney_datasets.py) using the NormalizationTransformer, before doing the train-valid-test split.

**Comment by VIGS25** [comment]
Closing this, since this has already been discussed and I am working on this now.

**Comment by chf42** [comment]
@VIGS25 I revisited this post. and noticed you were using "transform_y=True" in your transforms. I just wonder if I would like to do all feature normalization, should I use "transform_X = True" instead?

"  transformers = [
      deepchem.trans.NormalizationTransformer(
          transform_y=True, dataset=train, move_mean=move_mean)
  ]"

**Comment by VIGS25** [comment]
Yes.

**Comment by chf42** [comment]
@VIGS25 Thank you a lot for your confirmation! I tried to load the data with the normalization for both training set and test set. 
transformers = [dc.trans.NormalizationTransformer(transform_X=True, dataset=train]
for transformer in transformers:
        train_set = transformer.transform(train)
        test_set = transformer.transform(test)

It works fine for the train set, but I came across errors with test set transformer

/Users/deepchem/deepchem/trans/transformers.py:146: RuntimeWarning: invalid value encountered in true_divide
X = np.nan_to_num((X - self.X_means) / self.X_stds)

It is not that clear to me, since both train set and test set will go through the normalization process with the same means and stds. 

Thank you!
